### PR TITLE
[All Hosts] (sso) add configuration for unified manifest

### PR DESCRIPTION
**Pull request by Rick-Kirkham**

### CODE REVIEW

File: docs/develop/convert-xml-to-json-manifest.md [hunk 1]
```diff
@@ -50,6 +50,11 @@ When you've added the files to the project, add **\<IconUrl\>** and **\<HighReso
 
 If your manifest has any **\<FunctionName\>** elements, make sure their values have fewer than 65 characters. The value of this element must exactly match the name of a function in a JavaScript or TypeScript file. If you change it in the manifest, be sure to change it in the code file too.
 
+### Ensure that your SSO add-in requests permissions 
+
+If your add-in uses Microsoft single sign-in, your add-in has a **\<Scopes\>** element that specifies the Azure Active Directory or Microsoft Graph permissions that the add-in needs. With the unified manifest, permissions must be requested in runtime code. Update your code as needed to request these permissions. The exact code depends on the architecture and authorization code libraries you are using. Typically, code requests permissions in a function that requests an access token.
```
Comment (davidchesnut) [inline docs/develop/convert-xml-to-json-manifest.md:55]:
```suggestion
If your add-in uses Microsoft single sign-on with the on-behalf-of (OBO) flow, your add-in has a **\<Scopes\>** element that specifies the Microsoft Graph or other API permissions that the add-in needs. With the unified manifest, permissions must be requested at runtime in the code. Update your code as needed to request these permissions. The exact code depends on the architecture and authorization code libraries you are using. Typically, code requests permissions in a function that requests an access token. 
```

Comment (davidchesnut) [reply docs/develop/convert-xml-to-json-manifest.md:55]: How do you request permission from the code?

File: docs/develop/sso-in-office-add-ins.md [hunk 2]
```diff
@@ -69,7 +69,24 @@ For more details about this process, see [Register an Office Add-in that uses SS
 
 ### Configure the add-in
 
-Add new markup to the add-in manifest.
+Your manifest must provide Office with certain information about how the add-in is registered in Microsoft Entra. The configurtion depends on which type of manifest the add-in uses.
```
Comment (davidchesnut) [inline docs/develop/sso-in-office-add-ins.md:72]:
```suggestion
Your manifest must provide Office with certain information about how the add-in is registered in Microsoft Entra ID. The configuration depends on which type of manifest the add-in uses. 
```

File: docs/develop/sso-in-office-add-ins.md [hunk 3]
```diff
@@ -69,7 +69,24 @@ For more details about this process, see [Register an Office Add-in that uses SS
 
 ### Configure the add-in
 
-Add new markup to the add-in manifest.
+Your manifest must provide Office with certain information about how the add-in is registered in Microsoft Entra. The configurtion depends on which type of manifest the add-in uses. 
+
+# [Unified manifest](#tab/jsonmanifest)
+
+There should be a "webApplicationInfo" property in the root of the manifest. It has a required child "id" property which must be set to the application ID (a GUID) of the add-in in Microsoft Entra. For SSO, it must also have a child "resource" property that is set to the URI of the add-in. This is the same **Application ID URI** (including the `api:` protocol) that you set when you registered the add-in with Microsoft Entra. The URI must end with the client ID specified in the "webApplicationInfo.id" property. The following is an example:
```
Comment (davidchesnut) [inline docs/develop/sso-in-office-add-ins.md:76]:
```suggestion
There should be a "webApplicationInfo" property in the root of the manifest. It has a required child "id" property which must be set to the application ID (a GUID) of the add-in in the Microsoft identity platform. For SSO, it must also have a child "resource" property that is set to the URI of the add-in. This is the same **Application ID URI** (including the `api:` protocol) that you set when you registered the add-in with the Microsoft identity platform. The URI must end with the client ID specified in the "webApplicationInfo.id" property. The following is an example:
```

File: docs/develop/sso-in-office-add-ins.md [hunk 4]
```diff
@@ -69,7 +69,24 @@ For more details about this process, see [Register an Office Add-in that uses SS
...
+
+```json
+"webApplicationInfo": {
+	"id": "a661fed9-f33d-4e95-b6cf-624a34a2f51d",
+	"resource": "api://addin.contoso.com/a661fed9-f33d-4e95-b6cf-624a34a2f51d"
+},
+```
+
+> [!NOTE]
+> Experienced add-in developers should note that, unlike the XML manifest, you don't specify the Azure Active Directory or Microsoft Graph permissions that the add-in needs in the unified manifest. Request permissions in your runtime code.
```
Comment (davidchesnut) [inline docs/develop/sso-in-office-add-ins.md:86]:
```suggestion
> There is no corresponding JSON entry for the **\<Scopes\>** element. Microsoft Graph and other API permissions are requested at runtime in your code.
```

**Review by davidchesnut — CHANGES_REQUESTED** [review]

File: docs/develop/convert-xml-to-json-manifest.md (hunk 1 above)
Comment (Rick-Kirkham) [reply docs/develop/convert-xml-to-json-manifest.md:55]: > How do you request permission from the code?

It depends on the framework being used. For example, in our [Office-Add-in-Microsoft-Graph-ASPNET](https://github.com/OfficeDev/Office-Add-in-samples/blob/main/Samples/auth/Office-Add-in-Microsoft-Graph-ASPNET/Office-Add-in-Microsoft-Graph-ASPNETWeb/Controllers/AzureADAuthController.cs) sample, it's done with the following C# code: 

```
 // Generate the parameterized URL for Azure login.
 string[] graphScopes = { "Files.Read.All", "User.Read" };
 var urlBuilder = clientApp.GetAuthorizationRequestUrl(graphScopes);
 urlBuilder.WithRedirectUri(loginRedirectUri.ToString());
 urlBuilder.WithAuthority(Settings.AzureADAuthority);
 urlBuilder.WithExtraQueryParameters("state=" + authState);
 var authUrl = await urlBuilder.ExecuteAsync(System.Threading.CancellationToken.None);
         
 // Redirect the browser to the login page, then come back to the Authorize method below.
 return Redirect(authUrl.ToString());
```

File: docs/develop/sso-in-office-add-ins.md (hunk 4 above)
Comment (Rick-Kirkham) [reply docs/develop/sso-in-office-add-ins.md:86]:
```suggestion
> Experienced add-in developers should note that, there is no unified manifest property corresponding to the **\<Scopes\>** element in the XML manifest. Microsoft Graph and other API permissions are requested at runtime in your code.
```

File: docs/develop/sso-in-office-add-ins.md [hunk 5]
```diff
@@ -69,7 +69,24 @@ For more details about this process, see [Register an Office Add-in that uses SS
...
+
+```json
+"webApplicationInfo": {
+	"id": "a661fed9-f33d-4e95-b6cf-624a34a2f51d",
+	"resource": "api://addin.contoso.com/a661fed9-f33d-4e95-b6cf-624a34a2f51d"
+},
+```
+
+> [!NOTE]
+> Experienced add-in developers should note that, there is no unified manifest property corresponding to the **\<Scopes\>** element in the XML manifest. Microsoft Graph and other API permissions are requested at runtime in your code.
```
Comment (davidchesnut) [inline docs/develop/sso-in-office-add-ins.md:86]: You don't have to actually request the scopes in your code do you? If so I don't know if we document how to do that.

**Review by davidchesnut — APPROVED** [review]
One question I added, but looks good!

File: docs/develop/sso-in-office-add-ins.md (hunk 5 above)
Comment (Rick-Kirkham) [reply docs/develop/sso-in-office-add-ins.md:86]: > You don't have to actually request the scopes in your code do you? If so I don't know if we document how to do that.

It's doc'd in our samples: 
It depends on the framework being used. For example, in our [Office-Add-in-Microsoft-Graph-ASPNET](https://github.com/OfficeDev/Office-Add-in-samples/blob/main/Samples/auth/Office-Add-in-Microsoft-Graph-ASPNET/Office-Add-in-Microsoft-Graph-ASPNETWeb/Controllers/AzureADAuthController.cs) sample, it's done with the following C# code:

 // Generate the parameterized URL for Azure login.
 string[] graphScopes = { "Files.Read.All", "User.Read" };
 var urlBuilder = clientApp.GetAuthorizationRequestUrl(graphScopes);
 urlBuilder.WithRedirectUri(loginRedirectUri.ToString());
 urlBuilder.WithAuthority(Settings.AzureADAuthority);
 urlBuilder.WithExtraQueryParameters("state=" + authState);
 var authUrl = await urlBuilder.ExecuteAsync(System.Threading.CancellationToken.None);
         
 // Redirect the browser to the login page, then come back to the Authorize method below.
 return Redirect(authUrl.ToString());
Here's how it's done in one of our MSAL based samples: [excel-create-worksheet-from-web-site](https://github.com/OfficeDev/Office-Add-in-samples/blob/2a9356ce6513088977156201e7a381dbb2469efd/Samples/excel-create-worksheet-from-web-site/WebApplication/App/authConfig.js#L56)

/**
 * Scopes you add here will be prompted for user consent during sign-in.
 * By default, MSAL.js will add OIDC scopes (openid, profile, email) to any login request.
 * For more information about OIDC scopes, visit: 
 * https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-permissions-and-consent#openid-connect-scopes
 */
const loginRequest = {
    scopes: ["User.Read Files.ReadWrite"]
};

**Pull request merged**